libnmstate/nm/bond.py

```diff
@@ -34,9 +36,19 @@
 )
 
 
-def create_setting(options):
+def create_setting(options, wired_setting):
     bond_setting = nmclient.NM.SettingBond.new()
+    bond_mode = options[Bond.MODE]
     for option_name, option_value in options.items():
+        if (
+            wired_setting
+            and bond_mode == BondMode.ACTIVE_BACKUP
+            and option_name == "fail_over_mac"
+            and option_value in (1, "1", "active")
+        ):
+            # When using fail_over_mac=1 with active-backup mode, MAC address
+            # should be ignored
+            wired_setting.props.cloned_mac_address = None
```

Comment:
Is this correct? What about this:

1) User creates bond without fail_over_mac but specifies a MAC address
2) User now changes the bond mode

With this change Nmstate will silently change the MAC address but I guess it should raise an error. IMHO ideally the user would need to specify something like an empty MAC address to give their consent to remove the MAC address.

Reply:
1. User creates bond without fail_over_mac but specifies a MAC address

Done.

2. User now changes the bond mode

It's `test_bond_mac_restriction_in_desire_mac_in_current`

When user specify MAC address in desire state, we raise exception, we don't support empty MAC address yet.

When user never specify the MAC address, we remove that MAC added to state by merging,
it's not changing the MAC, user still get its current MAC address, it just not saved to the
NM connection profile. And that's what they should expect when setting `fail_over_mac`.